### PR TITLE
CONTRIBUTING: `uv` is now used to generate our `requirements.txt` files

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,10 +35,9 @@ and Python log messages in the debug console under "Python Server".
 - Update the `ruff` version in the README.md
   - in the Base URLs
   - in "The extension ships with `ruff==...`"
-- Make sure you use Python 3.7 installed and as your default Python.
-- Run `python -m venv .venv` to create a venv and activate it.
-- Run `python -m pip install pip-tools` to install `pip-tools`.
-- Run `rm requirements.txt requirements-dev.txt` and then `just lock` to update `ruff` and `ruff-lsp`.
+- Make sure you have Python 3.7 installed
+- Run `uv venv --python 3.7 && source .venv/bin/activate` to create a Python 3.7 venv and activate it.
```
Comment (AlexWaygood) [inline CONTRIBUTING.md:39]: Unfortunately `uv` will not install Python 3.7 for you if you don't have it already installed; 3.7 is too old for `uv`

**Review by dhruvmanila — APPROVED** [review]
Thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (dhruvmanila) [reply CONTRIBUTING.md:39]: Ah yeah, I remember facing this

Comment (dhruvmanila) [reply CONTRIBUTING.md:39]: I think Python is only required to bundle `ruff-lsp` so it should be fine

Comment (AlexWaygood) [reply CONTRIBUTING.md:39]: Python 3.7 is also still installable by other means, so it's "okay", just makes it a bit more annoying to work with if you don't already have Python 3.7 installed locally. Luckily I do 😄

**Pull request merged**